rust_src/src/lisp.rs

```diff
@@ -761,37 +752,37 @@ mod deprecated {
         assert!(!FLOATP(Qnil));
     }
 
-    #[allow(non_snake_case)]
+    // TODO: Rustic way?
+    #[deprecated]
     pub fn NATNUMP(a: LispObject) -> bool {
-        INTEGERP(a) && 0 <= XINT(a)
+        a.is_integer() && 0 <= a.to_fixnum().unwrap()
     }
 
     #[test]
     fn test_natnump() {
         assert!(!NATNUMP(Qnil));
     }
 
-    #[allow(non_snake_case)]
-    #[allow(dead_code)]
-    pub fn XFLOAT(a: LispObject) -> LispFloatRef {
-        debug_assert!(FLOATP(a));
-        unsafe { a.to_float_unchecked() }
+    #[deprecated]
+    pub unsafe fn XFLOAT(a: LispObject) -> LispFloatRef {
```

Comment:
Nice catch, this is definitely an example of a function that should be `unsafe`.